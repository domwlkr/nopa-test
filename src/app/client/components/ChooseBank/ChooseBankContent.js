import React, { PropTypes } from 'react';
import { Button } from '../index';
import initialState from '../../redux/reducers/initialState.js';
import * as Paths from '../../constants/paths';
import BankList from './BankList';

const bankList = initialState.bankList;

const ChooseBankContent = (props) => {
  return (
    <div className="main-content">
      <h1>Which bank does this account belong to?</h1>
      <p>Track all of your payments by connecting as many bank accounts as you'd like to your Nopa account and get updates on your balance instantly.</p>

      <BankList bankList={bankList} selectedBank={props.selectedBank} onSelectBank={props.onSelectBank} />

      <div className="error">{props.error}</div> 

      <Button onClick={props.onContinue} className="button">Continue</Button>

    </div>
  );
};

ChooseBankContent.propTypes = {
  bankList: PropTypes.array.isRequired,
  onSelectBank: PropTypes.func.isRequired,
  selectedBank: PropTypes.object.isRequired,
  error: PropTypes.string.isRequired,
  onContinue: PropTypes.func.isRequired
};

export default ChooseBankContent;