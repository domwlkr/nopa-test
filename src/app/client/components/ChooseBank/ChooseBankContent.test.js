import React from 'react';
import ChooseBankContent from './ChooseBankContent';
import initialState from '../../redux/reducers/initialState.js';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const bankList = initialState.bankList;
const onSelectBank = jest.fn();
const selectedBank = {name: 'Barclays', logo: 'test-file-stub'};
const onContinue = jest.fn();
const error = 'Error message';

test('Renders correctly', () => {
  const chooseBankContent = renderer.create(
    <ChooseBankContent bankList={bankList} onSelectBank={onSelectBank} error={error} onContinue={onContinue} selectedBank={selectedBank} />
  ).toJSON();

  expect(chooseBankContent).toMatchSnapshot();
});

test('Displays error message', () => {
  const chooseBankContent = shallow(
    <ChooseBankContent bankList={bankList} onSelectBank={onSelectBank} error={error} onContinue={onContinue} selectedBank={selectedBank} />
  );

  expect(chooseBankContent.instance().props.error).toEqual('Error message');
});

test('When continue button is clicked event fired', () => {
  const chooseBankContent = shallow(
    <ChooseBankContent bankList={bankList} onSelectBank={onSelectBank} error={error} onContinue={onContinue} selectedBank={selectedBank} />
  );

  chooseBankContent.find('Button').simulate('click');

  expect(onContinue).toHaveBeenCalled();
});