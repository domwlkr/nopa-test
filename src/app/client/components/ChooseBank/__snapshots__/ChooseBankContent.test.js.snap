// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="main-content"
>
  <h1>
    Which bank does this account belong to?
  </h1>
  <p>
    Track all of your payments by connecting as many bank accounts as you'd like to your Nopa account and get updates on your balance instantly.
  </p>
  <ul
    className="bank-list"
  >
    <li
      className="bank-item "
      onClick={[Function]}
    >
      <img
        alt="Barclays"
        className="bank-image"
        src="test-file-stub"
      />
    </li>
    <li
      className="bank-item "
      onClick={[Function]}
    >
      <img
        alt="Natwest"
        className="bank-image"
        src="test-file-stub"
      />
    </li>
    <li
      className="bank-item "
      onClick={[Function]}
    >
      <img
        alt="Lloyds"
        className="bank-image"
        src="test-file-stub"
      />
    </li>
    <li
      className="bank-item "
      onClick={[Function]}
    >
      <img
        alt="HSBC"
        className="bank-image"
        src="test-file-stub"
      />
    </li>
    <li
      className="bank-item "
      onClick={[Function]}
    >
      <img
        alt="TSB"
        className="bank-image"
        src="test-file-stub"
      />
    </li>
    <li
      className="bank-item "
      onClick={[Function]}
    >
      <img
        alt="Santander"
        className="bank-image"
        src="test-file-stub"
      />
    </li>
  </ul>
  <div
    className="error"
  >
    Error message
  </div>
  <a
    className="button"
    onClick={[Function]}
    style={Object {}}
  >
    Continue
  </a>
</div>
`;
